WIP: Looking For Group system update

Zerixx opened the pull request:
As the title says this is a work in progress. 

Goals: 
- Add, update, and implement packet structures.
- Update old methods and remove deprecated and or un-useful code.
- Add new technology to the LFG/DF system. E.g., Premade Finder.
- Make sure it works properly.
- More to be added as we get to it, or it's thought up.

Do not merge the pull request just yet.

I would LOVE it if other people with the similar interest in this system would like to contribute to the progress. It's a shame to see a great system no longer supported and updated for the current patch, or even expansion, at this point. 

If you notice any mistakes, errors, or issues, I would appreciate it if you could point them out to me, and if possible a solution. And remember, this is a WIP, and in NO MEANS is it functional, or working as intended. 

-Z

Commit:
Core/LFG: Added all the LFG and DF packets.

Re workings of LFG system coming at  a later date. This is mainly just for the sake of having these packets here.

Commit:
Core/PacketIO/LFG:

Fixed build.
Implemented ClientLFGJoinResult.

Commit:
Core/PacketIO: Updated some more LFG Opcodes

Commit:
Core/PacketIO: Fixed my derps. Now it builds. Getting too late.

Runico commented on src/server/game/Server/Packets/LFGPackets.cpp:
```diff
@@ -36,3 +36,354 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
+
+ByteBuffer& operator>>(ByteBuffer& data, WorldPackets::LFG::LfgBootInfo& bootInfo)
+{
+    data >> bootInfo.VoteInProgress;
```
typo ? >> should be << ?

Zerixx replied:
You may be right. Not at the computer ATM, will check when I can. I don't remember. It's like 5 am lmao.

Warpten replied:
wat

Zerixx replied:
No that I'm not completely incoherent, I realize what you mean. I will push an update soon for it. Don't need that method at all.

Commit:
Core/PacketIO:

- Updated a lot of the placeholder packet structures to actually  be correct.
- Updated some packet structures that were not already added.
- Removed some of the ByteBuffer operator methods which would've gone unused. Mostly for clientpackets.

Commit:
Core/PacketIO: Finished the packet structs.

MitchesD commented on src/server/game/Server/Packets/LFGPackets.cpp:
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
+        _worldPacket << Members;
+
+    _worldPacket.WriteBit(IsBeginning);
+    _worldPacket.WriteBit(ShowRoleCheck);
```
flush bits

MitchesD commented on src/server/game/Server/Packets/LFGPackets.cpp:
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
+    _worldPacket << Request;
+    _worldPacket << Unk;
+    _worldPacket << Reason;
+    _worldPacket.WriteBit(Listed);
```
flush

MitchesD commented on src/server/game/Server/Packets/LFGPackets.cpp:
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
+        _worldPacket << Players;
+
+    _worldPacket.WriteBit(ValidCompletedMask);
+    _worldPacket.WriteBit(ProposalSilent);
```
flush

MitchesD commented on src/server/game/Server/Packets/LFGPackets.cpp:
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
+
+    _worldPacket << Removes;
+
+    _worldPacket.WriteBit(Incremental);
```
flush

MitchesD commented on src/server/game/Server/Packets/LFGPackets.cpp:
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
+
+    for (uint32 i = 0; i < slotsCount; i++)
+        _worldPacket >> Slots;
+}
```
no new line at the end of file

MitchesD commented:
update these structure is easy, but it's pain to rewrite whole LFG system and implement new needed things, already did some of this, will share something later with you. To sum up I think this is about 1% from the whole patch which you will have to do.

Warpten commented on src/server/game/Server/Packets/LFGPackets.cpp:
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
+        data << playerDungeonInfo.ShortageReward;
+
+    data.WriteBit(playerDungeonInfo.FirstReward);
+    data.WriteBit(playerDungeonInfo.ShortageEligible);
```
missing flush

Warpten commented on src/server/game/Server/Packets/LFGPackets.cpp:
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
+    data << playerRewards.RewardItemQuantity;
+    data << playerRewards.BonusCurrency;
+    data.WriteBit(playerRewards.IsCurrency);
+
```
here too

Warpten commented on src/server/game/Server/Packets/LFGPackets.cpp:
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
+
+    data.WriteBit(playerQuestRewards.UnkBit);
+
+    if (playerQuestRewards.UnkBit)
```
required flush here too!

Warpten commented on src/server/game/Server/Packets/LFGPackets.cpp:
```diff
@@ -36,3 +36,553 @@ ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::LFG::RideTicket const& ti
+    data.WriteBit(update.SameParty);
+    data.WriteBit(update.Responded);
+    data.WriteBit(update.Accepted);
+
```
take a guess

Warpten commented:
Packet names must not contain the UserClient, Client, etc... prefixes.

Zerixx commented:
@MitchesD @Warpten I hadn't commit an update just yet, but I took care of the FlushBits already. The only reason I hadn't sent in the fixes is because I'm working on handlers right now, and didn't wanna commit seriously broken and nasty code.

However, anything you've got to help would go a long long way. Warpten, I didn't realize that I can't use those prefixes. Will get those changed asap.

Commit:
Core/PacketIO: Fixed the packet names and the FlushBits.

Left out the changes to the system because, frankly, they're not done and poorly written right now.

Commit:
Core/PacketIO: Fixed the packet names and the FlushBits. Pt.2

Left out the changes to the system because, frankly, they're not done and poorly written right now.

Commit:
Core/PacketIO: Forgot to commit these updates.

Thanks to @MitchesD for the help.

spuresz commented:
trinitycore when it will be loaded?

MitchesD commented:
@Zerixx any news? :)

Zerixx commented:
@MitchesD Yes! I am still working on it, because I have been incredibly busy recently. But, I am happy to say that I'm getting closer and closer. I deleted the old repo, and made a new one, so I will eventually resubmit a new PR!

tkrokli commented:
@Zerixx : do you think you can update your git setup with the e-mail address of your github account?
It looks like your commits don't link to your account because some ID info is missing in your git setup.

Zerixx commented:
@tkrokli Yeah that was fixed a while back, and I also had too much garbage in that repo so I just did it over.

MitchesD commented:
Closing this PR, he created new branch and will open new PR when it's ready

Voidx commented:
lfg system is pretty broken. This system stop work every 2 or 3 hours.
